gcloud/datastore/helpers.py

```diff
@@ -33,6 +33,46 @@
 INT_VALUE_CHECKER = Int64ValueChecker()
 
 
+def _find_true_dataset_id(dataset_id, connection):
```

Comment:
This still feels to me as though it should be public:  an application developer who wanted to be "correct" might call it once at startup, and then use the normalized value thereafter when creating keys, knowing that she could safely compare them with those returned from the back-end in other API calls.